README.ideal.md

```diff
@@ -0,0 +1,131 @@
+_Ni-Mh — batteries included environments with Nickel inside_.
+
+Managing a project's development environment involves configuring a lot of tools and occasionnally getting to communicate with each other: One (or several) package manager, a CI system, a service manager (to run that local postgresql database without which you can't test anything), custom utility scripts that end up spawning everywhere, etc.
+Ni-Mh (called “nickel-nix” or “Nixel” at the moment, but I really like Ni-Mh) aims at being your main entrypoint for managaing all these different tools, so that you can get:
+
+1. A unified configuration framework for all of these
+2. A powerful and ergonomic language to allow you to easily abstract over these configurations, with discoverability and early error reporting
+3. A way to get these tools to communicate with each others, so that your custom scripts can easily leverage your package manager to get the dependencies they need, your CI setup can understand your build system and parallelise efficiently, and so on.
+
+## Getting started
+
+To start using `nickel-nix`, you need `Nix` to be installed and configured.
+If it isn't already the case, you can get it with:
+
+```console
+$ curl -L https://nixos.org/nix/install | bash
+# We also need a couple of experimental Nix features
+$ mkdir -p ~/.config/nix
+$ echo 'experimental-features = nix-command flakes' >> ~/.config/nix/nix.conf
+```
+
+Then bootstrap a project with:
+
+```console
+$ nix flake init -t github:nickel-lang/nickel-nix
+# Edit the project file to fit your needs
+$ $EDITOR project.ncl
+# Enter the environment
+$ . .nickel-nix/activate
```

Comment:
Good question. Keeping in mind that this is an ideal goal, there are two things that are annoying with `nix develop`:

1. The model of “entering a new (bash) shell” is not very friendly – especially for people who don't use bash usually, but not only. A direnv-like model of injecting stuff into the editor is less pure, but nicer;
2. We might want to do some extra stuff (regenerating lockfiles and such) before calling Nix, in which case sourcing our own script is pretty much mandatory.

Reply:
I think that it's hard to consider sourcing a file as being ideal (plus, sourcing a file depends on your shell, so there is some difficulty there).

But at any rate, I'd restrict this README to current targets (otherwise, we could just have a single line README “you already use it”). If our current target is to use `nix develop`, let's avoid making the README confusing (if someone contributes the `activate` script, they'd seem to be following the roadmap, if it's not in the roadmap right now, then let's not have it in the README). If we still believe that Nixel ought to be lightweight to the external contributor in order to succeed in the open source word, I'd very much like to avoid such unfamiliar instructions.

For a direnv-like model, let's let direnv do the job.

For (2), though, I don't necessarily have a solution. But then again: we're writing an ideal README, so we're probably allowed to make promises that we don't know how to keep.

Reply:
Fair enough, I'll keep `nix develop` here, we'll see whether it's enough in the long run